tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+"""Unit tests for the ``repositories`` paths."""
+from ddt import ddt
+from fauxfactory import gen_choice, gen_string, gen_url
+from nailgun import client
+from random import randint, shuffle
+from requests.exceptions import HTTPError
+from robottelo import entities
+from robottelo.common.constants import DOCKER_REGISTRY_HUB
+from robottelo.common.decorators import (
+    data, run_only_on, skip_if_bug_open, stubbed)
+from robottelo.common.helpers import get_server_credentials
+from robottelo.test import APITestCase
+# (too-many-public-methods) pylint:disable=R0904
+
+
+STRING_TYPES = ['alpha', 'alphanumeric', 'cjk', 'utf8', 'latin1']
+
+
+def _create_repository(prod_id, name=None, upstream_name=None):
+    """Creates a Docker-based repository.
+
+    :param str name: Name for the repository. If ``None`` then a random
+    value will be generated.
+    :param str upstream_name: A valid name for an existing Docker image.
+    If ``None`` then defaults to ``busybox``.
+
+    :return: A dictionary representing the created repository.
+
+    """
+
+    if name is None:
+        name = gen_string(gen_choice(STRING_TYPES), 15)
+    if upstream_name is None:
+        upstream_name = u'busybox'
+
+    return entities.Repository(
+        product=prod_id,
+        content_type=u'docker',
+        name=name,
+        docker_upstream_name=u'busybox',
+        url=DOCKER_REGISTRY_HUB
+    ).create()
```

Comment:
`s/create/create_json`

Someday, eventually, `create` will return a `Repository` object with all of its attributes populated, and `create_json` will return a dict.